addons/enemybar/enemybar.lua

```diff
@@ -24,31 +24,524 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+            else
+                windower.add_to_chat(123, 'EnemyBar: no target selected to focustarget')
+            end
+        end
+    elseif S{'demo','setup','debug','test'}:contains(c) then
+        if args[3] then
+            state.setup = normalize_boolean(args[3])
+        else
+            state.setup = not state.setup
+        end
+        windower.add_to_chat(207, 'EnemyBar: setup mode is now "'..(state.setup and 'on' or 'off')..'"')
+    elseif S{'help','h','man','manual'}:contains(c) then
+        helptext = [[Enemy Bar - Command List:')
+1. set/s [setting] [target/t/subtarget/st/aggro/a/all] [value] - set a setting to its value
+    setting: pos(x y)/font/font_size/color(r g b)/width/count/show/show_target_icon/show_debuff/show_dist/show_action/show_target
+2. focustarget/ft/f (player_name or id or blank or clear) - create a bar for a particular party member, mob by ID, or by current target (blank), or clear the current focus target
+3. setup/demo/debug/test - toggles setup mode displaying test versions of all options and enabling drag for each frame
+4. help/h/manual/man --Shows this menu.]]
+        for _, line in ipairs(helptext:split('\n')) do
+                windower.add_to_chat(207, line)
+        end
+    end
+end
+
+function get_mob_by_name(n)
+    n = n:lower()
+    local worse_match = nil
+    local worser_match = nil
+    for _,mob in pairs(windower.ffxi.get_mob_array()) do
+        local mobname = mob.name:lower()
+        if n == mobname then
+            return mob
+        elseif not worse_match and mobname:sub(1, #n) == n then
+            worse_match = mob
+        elseif not worser_match and mobname:match(n) then
+            worser_match = mob
+        end
+    end
+    return worse_match and worse_match or worser_match
+end
+
+function normalize_bar_name(n)
+    n = n:lower()
+    if n == 't' then
+        n = 'target'
+    elseif n == 'st' then
+        n = 'subtarget'
+    elseif n == 'a' then
+        n = 'aggro'
+    elseif S{'f','ft'}:contains(n) then
+        n = 'focustarget'
+    elseif not S{'target','subtarget','aggro','focustarget','all'}:contains(n) then
+        n = nil
+    end
+
+    return n
+end
+
+function normalize_boolean(b)
+    b = b:lower()
+    if S{'true','t','yes','y','on'}:contains(b) then
+        return true
+    elseif S{'false','f','no','n','off'}:contains(b) then
+        return false
+    else return nil end
+end
+
+function set_setting(bar, setting, v)
+    if bar == 'all' then
+        settings['target_bar'][setting] = v
+        settings['subtarget_bar'][setting] = v
+        settings['aggro_bar'][setting] = v
+        settings['focustarget_bar'][setting] = v
+    else
+        settings[bar..'_bar'][setting] = v
+    end
+    windower.add_to_chat(207, 'EnemyBar: "'..setting..'" updated for "'..bar..'" bar')
+    settings:save()
+    initialize_bars()
+end
+
+function load_settings()
+    defaults = {}
```

Comment:
The `config` library has a feature where it reloads settings based on a new character itself. You do not need to define this function. Just define the defaults globally and do `local settings = config.load(defaults)`. After a logout/login this object will already contain updated values based on the settings of the new character. See also my note below about `config.register`.